src/main/java/htsjdk/samtools/BamFileIoUtils.java

```diff
@@ -21,7 +22,11 @@
 public class BamFileIoUtils {
     private static final Log LOG = Log.getInstance(BamFileIoUtils.class);
 
-    public static final String BAM_FILE_EXTENSION = "." + SamReader.Type.BAM_TYPE.fileExtension();
+    /**
+     * @deprecated Use {@link IOExtensions#BAM_FILE_EXTENSION} instead.
```

Comment:
Please add the date (month & year to the deprecation notices to make actually removing the code easier)

Reply:
I've added dates to all deprecations:
```/**
     * @deprecated since June 2019 Use {@link FileExtensions#VCF} instead.
     */
    @Deprecated
    public static final String VCF_FILE_EXTENSION = FileExtensions.VCF;
```

I've also changed internal usages of the deprecated code over to the new code.

Reply:
I've added 3 more file extensions to `FileExtensions`: `.bai, .crai, .csi` and added deprecation notices, changed over internal usages, etc for these 3.